src/Extensions/UsedOnTableExtension.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace SilverStripe\LinkField\Extensions;
+
+use DNADesign\Elemental\Models\BaseElement;
+use SilverStripe\Core\Extension;
+use SilverStripe\LinkField\Models\FileLink;
+use SilverStripe\ORM\DataObject;
+
+class UsedOnTableExtension extends Extension
+{
+    public function updateUsageAncestorDataObjects(array &$ancestorDataObjects, DataObject $dataObject): void
+    {
+        if (!is_a($dataObject, FileLink::class)) {
+            return;
+        }
+        $owner = $dataObject->Owner();
+        $ancestorDataObjects[] = $owner;
+        if (class_exists(BaseElement::class)
+            && is_a($owner, BaseElement::class)
+            && $owner->hasMethod('getTopPage')
+        ) {
+            $page = $owner->getTopPage();
+            if ($page) {
```

Comment:
```suggestion
            if ($page && $page->exists()) {
```
Note that while the `exists()` call is currently redundant, it's good practice and protects against possible changes in the future.

Reply:
Should just deprecate exists() then?

Reply:
Updated